src/LinuxTracing/TracerImpl.cpp

```diff
@@ -1041,6 +1079,23 @@ uint64_t TracerImpl::ProcessSampleEventAndReturnTimestamp(const perf_event_heade
     DeferEvent(event);
     ++stats_.uprobes_count;
 
+  } else if (is_uprobe_with_stack) {
+    pid_t pid = ReadSampleRecordPid(ring_buffer);
+    const size_t size_of_uprobe_sample = sizeof(perf_event_sp_stack_user_sample_fixed) +
+                                         2 * sizeof(uint64_t) /*size and size*/ +
+                                         stack_dump_size_ /*data*/;
+    if (header.size != size_of_uprobe_sample) {
+      ring_buffer->SkipRecord(header);
+      return timestamp_ns;
+    }
+    if (pid != target_pid_) {
+      ring_buffer->SkipRecord(header);
+      return timestamp_ns;
+    }
+
+    UprobesWithStackPerfEvent event = ConsumeUprobeWithStackPerfEvent(ring_buffer, header);
+    DeferEvent(std::move(event));
+    ++stats_.uprobes_count;
```

Comment:
This is a bit of a different statistics. Either add a new field for this or don't increment it.